### ISSUE TITLE
Error message not correct

### ISSUE THREAD

**Issue by benknoble** [abrahampost/Butter#6]
Sample run
```
$ ./Butter
Butterv0.1 (repl)
> string a := "a"
> a
a
> a := 5
RUNTIME_ERROR: Cannot assign value to int type
```

**Comment by benknoble** [comment]
Looks like method `assign` in `environment.go`: perhaps we should just verify that `value.(type)` and `found.(type)` are the same.

At any rate, the error message seems backwards. I would have expected "Cannot assign int to string type" or something like that.

**Comment by abrahampost** [comment]
Yep, that makes sense. I will try to submit a fix later tonight!

The reason the logic is a little squirrely currently is that I haven't quite gotten the hang of Go's type system yet. You can only use \<variable\>.(type) in a switch statement, so I was struggling with type checking in a few places. I think I will try to use one of the techniques outlined in this [stack overflow post](https://stackoverflow.com/questions/20170275/how-to-find-a-type-of-an-object-in-go) in order to simplify the logic a bit.

Thanks a bunch for finding that error!!!

**Comment by abrahampost** [comment]
closed in #7
